containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -0,0 +1,216 @@
+    def from_arn(arn):
+        if not arn.startswith("s3://"):
+            raise ValueError(arn)
+
+        bucket, prefix = arn.replace("s3://", "").split("/", 1)
+
+        return BucketPath(bucket, prefix)
+
+
+def get_bucket_name() -> str:
+    """
+    Get the bucket name from the environment
+    """
+    return os.environ["BUCKET_NAME"]
+
+
+def get_raw_database_name() -> str:
+    """
+    The athena database for raw data
+    """
+    return "data_products_raw"
+
+
+def get_raw_table_name(table_prefix) -> str:
+    """
+    The athena table for raw data
+    """
+    return f"{table_prefix}_raw"
+
+
+def get_extraction_timestamp(raw_data: BucketPath):
+    """
+    Parse extraction timestamp from the raw data path
+    """
+    match = EXTRACTION_TIMESTAMP_REGEX.match(raw_data.prefix)
+
+    if not match:
+        raise ValueError(
+            "Table partition extraction_timestamp is not in the expected format"
+        )
+
+    return datetime.strptime(match.group(3), EXTRACTION_TIMESTAMP_FORMAT)
+
+
+def data_product_raw_data_file_path(
+    database_name: str,
+    table_name: str,
+    extraction_timestamp: datetime,
+    uuid_value: uuid,
+    bucket_name: str = None,
+) -> BucketPath:
+    if bucket_name is None:
+        bucket_name = get_bucket_name()
+
+    amz_date = extraction_timestamp.strftime(EXTRACTION_TIMESTAMP_FORMAT)
+    prefix = os.path.join(
+        "raw_data",
+        database_name,
+        table_name,
+        f"extraction_timestamp={amz_date}",
+        str(uuid_value),
+    )
+
+    return BucketPath(bucket=bucket_name, prefix=prefix)
+
+
+def data_product_curated_data_path(
+    database_name: str, table_name: str, bucket_name: str = None
+) -> BucketPath:
+    """
+    The S3 location for partitioned data files in parquet format.
+    """
+    if bucket_name is None:
+        bucket_name = get_bucket_name()
+
+    prefix = (
+        os.path.join(
+            "curated_data",
+            f"database_name={database_name}",
+            f"table_name={table_name}",
+        )
+        + "/"
+    )
+
+    return BucketPath(bucket_name, prefix)
+
+
+def data_product_metadata_path(
+    data_product_name: str, bucket_name: str = None
+) -> BucketPath:
+    """
+    Generate the metadata path based on the data product name
+    """
+    if bucket_name is None:
+        bucket_name = get_bucket_name()
+
+    metadata_s3_path = os.path.join(
+        "metadata",
+        data_product_name,
+        "v1.0",
+        "metadata.json",
+    )
+
+    return BucketPath(bucket=bucket_name, prefix=metadata_s3_path)
+
+
+@dataclass
+class DataProductPathConfig:
+    """
+    Configures the names and paths for a data product.
+    """
+
+    extraction_timestamp: datetime  # Date the raw data was gathered
+    account_id: str  # AWS account ID
+
+    # S3 paths
+    raw_data_file: BucketPath
+    curated_data_path: BucketPath
+
+    # Athena database and table names
+    database_name: str
+    table_name: str
```

Comment:
I think this should be replaced with `data_product_name`.

If we're talking about database_name and table_name it should specify whether we are talking about raw or curated data.